mssqlcli/mssqlqueries.py

```diff
@@ -164,3 +175,13 @@ def get_foreignkeys():
             AND kcu2.constraint_name = rc.unique_constraint_name
             AND kcu2.ordinal_position = kcu1.ordinal_position
             ORDER BY 3, 4'''
+    return normalize(sql)
+
+
+def normalize(sql):
+    if (sql == '' or sql is None):
+        return sql
+    else:
+        sql = sql.replace('\r', ' ').replace('\n', ' ').strip()
+        sql = re.sub(r' +', ' ', sql)
```

Comment:
we didn't neet these replace and substitutions before. Why now?

Reply:
I noticed some of our code recognized the type of command by looking at a word before the first space character. For example:
```
# mssqlcli\packages\special\main.py

def parse_special_command(sql):

    # split the sql by space -- 'command' is a word before first space.
    command, _, arg = sql.partition(' ')

    verbose = '+' in command
    command = command.strip().replace('+', '')
    return (command, verbose, arg.strip())
```

This assumes the word before the first space character is keyword of command (very fragile). Unfortunately we have methods that returns sql with bunch of leading white spaces. For example:
```
def get_sql():
    return '''
        SELECT col1,
               col2
        FROM t1
        '''
```
The method returns a string:
```
'\n        SELECT col1,\n               col2\n        FROM t1\n        '
```
The word before the first space is `\n`, which is recognized as a command keyword. This not intended.
I understand the original author used ''' type string for free-formatting, but I believe these methods have to return correctly formatted such as:
```
'SELECT col1, col2 FROM t1'
```
The `normalize()` method I wrote converts the raw format into correct format.